hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java

```diff
@@ -0,0 +1,176 @@
+            }
+        }
+        // Mold the fungible changes into a transfer ordered by (token ID, account ID). The fungible pairs are ordered
+        // by (accountId, tokenId), so we need to group by each token ID
+        for (final var acctAmountsForToken : acctAmountsByTokenId.entrySet()) {
+            final var singleTokenTransfers = acctAmountsForToken.getValue();
+            if (!singleTokenTransfers.isEmpty()) {
+                final var aaList = asAccountAmounts(singleTokenTransfers);
+                aaList.sort(ACCOUNT_AMOUNT_COMPARATOR);
+                fungibleTokenTransferLists.add(TokenTransferList.newBuilder()
+                        .token(acctAmountsForToken.getKey())
+                        .transfers(aaList)
+                        //                        .expectedDecimals(0) // Should this be set ?
+                        .build());
+            }
+        }
+
+        return fungibleTokenTransferLists;
+    }
+
+    @NonNull
+    protected Map<TokenID, List<NftTransfer>> nftChangesFrom(@NonNull final WritableNftStore writableNftStore) {
+        final var nftChanges = new HashMap<TokenID, List<NftTransfer>>();
+        for (final NftID nftId : writableNftStore.modifiedNfts()) {
+            final var modifiedNft = writableNftStore.get(nftId);
+            final var persistedNft = writableNftStore.getOriginalValue(nftId);
+
+            // The NFT may not have existed before, in which case we'll use a null sender account ID
+            final var senderAccountId = persistedNft != null ? persistedNft.ownerId() : null;
+            final var nftTransfer = NftTransfer.newBuilder()
+                    .serialNumber(modifiedNft.id().serialNumber())
+                    .senderAccountID(senderAccountId)
+                    .receiverAccountID(modifiedNft.ownerId())
+                    .isApproval(false) // Isn't this wrong ? Should we see transaction and set isApproval ?
+                    .build();
+            if (!nftChanges.containsKey(nftId.tokenId())) {
+                nftChanges.put(nftId.tokenId(), new ArrayList<>());
+            }
+
+            final var currentNftChanges = nftChanges.get(nftId.tokenId());
+            currentNftChanges.add(nftTransfer);
+            nftChanges.put(nftId.tokenId(), currentNftChanges);
+        }
+        return nftChanges;
+    }
+
+    protected List<TokenTransferList> asTokenTransferListFromNftChanges(
+            final Map<TokenID, List<NftTransfer>> nftChanges) {
+
+        // Create a new transfer list for each token ID
+        final var nftTokenTransferLists = new ArrayList<TokenTransferList>();
+        for (final var nftsForTokenId : nftChanges.entrySet()) {
+            if (!nftsForTokenId.getValue().isEmpty()) {
+                // This var is the collection of all NFT transfers _for a single token ID_
+                final var nftTransfersForTokenId = nftsForTokenId.getValue();
+                nftTransfersForTokenId.sort(NFT_TRANSFER_COMPARATOR);
```

Comment:
🤔 I just checked and `mono-service` **doesn't** order NFT transfers by serial number. 

Instead, the order in the record matches the order in the `CryptoTransferTransactionBody` transfer list.

Reply:
I see removed comparison